src/lib.rs

```diff
@@ -210,6 +218,14 @@ impl Tokenizer {
         out.reserve(text.as_bytes().len());
         let words = self.word_split.find_iter(&text).map(|m| m.as_str());
         for word in words {
+            if word == "<start_of_text>" {
+                out.push(self.start_of_text());
+                continue;
+            }
+            if word == "<end_of_text>" {
```

Comment:
I think using `else if` here makes sense? Slightly more compact and encodes the notion that these branches are mutually exclusive.